timeseries-bucket-repair/README.md

```diff
@@ -0,0 +1,118 @@
+# Repair internal Buckets in Time Series Collections
+
+## Warning
+
+The scripts provided here are best run by those experienced with MongoDB, ideally with guidance from MongoDB Technical Support. They are provided without warranty or guarantee of any kind (see disclaimer below). If you do not have access to MongoDB Technical Support, consider reaching out to our community at [community.mongodb.com](community.mongodb.com) to ask questions.
+
+Some adaptation of these scripts may be required for your use-case.
+
+If you are using these scripts on your own, we strongly recommend:
+
+* reading all instructions before attempting any action, and ensuring you are prepared for all steps.
+* taking regular backups of the [dbpaths](https://docs.mongodb.com/manual/core/backups/#back-up-by-copying-underlying-data-files) of each node, and working off of backups as much as possible.
+* reviewing the scripts themselves to understand their behavior.
+* testing this process and all scripts on a copy of the environment it is to be run.
+* for sharded clusters, disabling the balancer.
+
+# Prerequisites 
+- This script should be run with a user that has [dbAdmin](https://www.mongodb.com/docs/manual/reference/built-in-roles/#mongodb-authrole-dbAdmin) permissions on the database(s) for the affected time-series collection(s).
+
+# Determine if You're Impacted
+
+Users can determine if they have been impacted by running [`validate`](https://www.mongodb.com/docs/manual/reference/command/validate/) on their Time Series collections and checking the `validate.errors` and `validate.warnings` fields to determine if there are problems with time series buckets detected.
```

Comment:
Would it be helpful to link to CAs where this could be a solution? Or bugs/tickets that are related to this that are public?

Reply:
Following-up on this, was there testing to see if this would correct the specific use-cases that this script is supposed to solve?

Reply:
I feel this is the same feedback as Dan provided here: https://github.com/mongodb/support-tools/pull/151#discussion_r2162624029

I kind of want this to be a generic script to mimic our logical initial sync capability for regular collection to perform a logical rewrite.